course_catalog/tasks.py

```diff
@@ -122,6 +123,43 @@ def import_all_ocw_files(self, chunk_size):
     raise self.replace(tasks)
 
 
+@app.task
+def get_xpro_files(ids):
+    """
+    Task to sync xPRO OLX course files with database
+    """
+    if not (
+        settings.XPRO_LEARNING_COURSE_BUCKET_NAME
+        and settings.XPRO_LEARNING_COURSE_ACCESS_KEY
+        and settings.XPRO_LEARNING_COURSE_SECRET_ACCESS_KEY
+    ):
+        log.warning("Required settings missing for get_xpro_files")
+        return
+    sync_xpro_course_files(ids)
+
+
+@app.task(bind=True)
+def import_all_xpro_files(self, chunk_size):
+    """Ingest xPRO OLX files from the S3 bucket"""
+    blacklisted_ids = load_course_blacklist()
+    tasks = celery.group(
+        [
+            get_xpro_files.si(ids)
+            for ids in chunks(
+                Course.objects.filter(published=True)
+                .filter(platform=PlatformType.xpro.value)
+                .exclude(course_id__in=blacklisted_ids)
```

Comment:
Should there be a filter for published?

Reply:
It's at line 149. It's a little confusing since there are two filters chained together.